ghrn/ghrn.go

```diff
@@ -13,96 +13,118 @@ import (
 
 // Config describes configuration for BuildReleaseNotes.
 type Config struct {
-  // Org is the name of the GitHub organization. Required.
-  Org string
-  // Repo is the name of the GitHub repository. Required.
-  Repo string
-
-  // GitHubToken is a GitHub API access token.
-  GitHubToken string
-
-  // StopAt is the number of the Pull Request to stop at.
-  // Useful for building the notes of PRs since the last release, for example.
-  StopAt int
-  // IncludeCommits will include commmits messages for each PR.
-  IncludeCommits bool
+	// Org is the name of the GitHub organization. Required.
+	Org string
+	// Repo is the name of the GitHub repository. Required.
+	Repo string
+
+	// GitHubToken is a GitHub API access token.
+	GitHubToken string
+
+	// StopAt is the number of the Pull Request to stop at.
+	// Useful for building the notes of PRs since the last release, for example.
+	StopAt int
+	// IncludeCommits will include commmits messages for each PR.
+	IncludeCommits bool
+	// StopAtLatestRelease will stop at latest release commit.
+	StopAtLatestRelease bool
 }
 
 // BuildReleaseNotes lists GitHub Pull Requests and writes formatted release notes
 // to the given writer.
 func BuildReleaseNotes(ctx context.Context, w io.Writer, conf Config) error {
 
-  if conf.Org == "" {
-    return fmt.Errorf("Config.Org is required")
-  }
-  if conf.Repo == "" {
-    return fmt.Errorf("Config.Repo is required")
-  }
-
-  var httpClient *http.Client
-  if conf.GitHubToken != "" {
-    ts := oauth2.StaticTokenSource(
-      &oauth2.Token{AccessToken: conf.GitHubToken},
-    )
-    httpClient = oauth2.NewClient(ctx, ts)
-  }
+	if conf.Org == "" {
+		return fmt.Errorf("Config.Org is required")
+	}
+	if conf.Repo == "" {
+		return fmt.Errorf("Config.Repo is required")
+	}
+
+	var httpClient *http.Client
+	if conf.GitHubToken != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: conf.GitHubToken},
+		)
+		httpClient = oauth2.NewClient(ctx, ts)
+	}
 	cl := github.NewClient(httpClient)
 
 	opt := &github.PullRequestListOptions{
 		ListOptions: github.ListOptions{PerPage: 100},
 		State:       "closed",
 	}
 
+	repo, _, err := cl.Repositories.Get(ctx, conf.Org, conf.Repo)
+	if err != nil {
+		return fmt.Errorf("get repository: %+v", err)
+	}
+
+	rls, _, err := cl.Repositories.GetLatestRelease(ctx, conf.Org, conf.Repo)
+	if err != nil {
+		return fmt.Errorf("get latest release: %+v", err)
+	}
+
+	comp, _, err := cl.Repositories.CompareCommits(ctx, conf.Org, conf.Repo, rls.GetTagName(), repo.GetDefaultBranch())
+	if err != nil {
+		return fmt.Errorf("compare commitse: %s..%s %+v", rls.GetTagName(), repo.GetDefaultBranch(), err)
+	}
```

Comment:
I think these are only needed for when conf.SinceLatestRelease is true. Can you move these to a helper, and only use them if we're `conf.SinceLatestRelease == true`? This would avoid a few calls to github and maybe save 1-2 seconds in some cases.